File: Java/Medium/CompileError/valid-square_2.java
class Solution {
    public boolean validSquare(int[] p1, int[] p2, int[] p3, int[] p4) {
        
        int[] d = new int[6];
        d[0] = dist(p1, p2);
        d[1] = dist(p1, p3);
        d[2] = dist(p1, p4);
        d[3] = dist(p2, p3);
        d[4] = dist(p2, p4);
        d[5] = dist(p3, p4);
        Arrays.sort(d);
        return d[0] > 0 && d[0] == d[1] && d[0] == d[2] && d[0] == d[3] && d[4] == d[5];
    }
}

File: Python3/Medium/WrongAnswer/valid-square_2.py
class Solution:
    def validSquare(self, p1: List[int], p2: List[int], p3: List[int], p4: List[int]) -> bool:
        
        
        def dist(x1, y1, x2, y2):
            return (x1 - x2) ** 2 + (y1 - y2) ** 2
        
        points = [p1, p2, p3, p4]
        points.sort()
        p1, p2, p3, p4 = points
        return dist(*p1, *p2) > 0 and dist(*p1, *p2) == dist(*p2, *p3) == dist(*p3, *p4) == dist(*p4, *p1) and dist(*p1, *p3) == dist(*p2, *p4)